packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -428,6 +444,101 @@ - (NSView *)hitTest:(NSPoint)point
   return isTextViewClick ? self : hitView;
 }
 
+- (NSNumber *)reactTagAtMouseLocationFromEvent:(NSEvent *)event
+{
+  NSPoint locationInSelf = [self convertPoint:event.locationInWindow fromView:nil];
+  NSPoint locationInInnerTextView = [self convertPoint:locationInSelf toView:_textView]; // This is needed if the parent <Text> view has padding
+  return [self reactTagAtPoint:locationInInnerTextView];
+}
+
+- (void)mouseEntered:(NSEvent *)event
+{
+  // superclass invokes self.onMouseEnter, so do this first
+  [super mouseEntered:event];
+
+  [self updateHoveredSubviewWithEvent:event];
+}
+
+- (void)mouseExited:(NSEvent *)event
+{
+  [self updateHoveredSubviewWithEvent:event];
+
+  // superclass invokes self.onMouseLeave, so do this last
+  [super mouseExited:event];
+}
+
+- (void)mouseMoved:(NSEvent *)event
+{
+  [self updateHoveredSubviewWithEvent:event];
+
+  // As of the time of writing (2024-06-25), the latest versions of Safari, Firefox, Chrome, and Edge
```

Comment:
This isn't accurate; mousemove events can be captured before the bubble phase.  For key, focus, and mouse events we've generally followed the [W3C Model](https://www.quirksmode.org/js/events_order.html#link4:~:text=W3C%20has%20very%20sensibly%20decided%20to%20take%20a%20middle%20position%20in%20this%20struggle.%20Any%20event%20taking%20place%20in%20the%20W3C%20event%20model%20is%20first%20captured%20until%20it%20reaches%20the%20target%20element%20and%20then%20bubbles%20up%20again.) of a capture phase that routes from root to target and a bubble phase that routes from target to root.

Yes there is definitely some complexity in the W3C spec if we look at the details of e.g. [mouseleave vs mouseout](https://www.w3.org/TR/uievents/#mouseleave:~:text=This%20event%20type%20is%20similar%20to%20mouseout%2C%20but%20differs%20in%20that%20does%20not%20bubble) but we aren't solving for that right now :)

Reply:
You know what, this comment just seems way too confusing. Probably better to remove it since we don't even support general mouse move events on the JS side anyway.